python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -453,6 +457,27 @@ def measureForcedSources(self, diaSources, science, wcs):
         for diaSource, forcedSource in zip(diaSources, forcedSources):
             diaSource.assign(forcedSource, mapper)
 
+    def calculateMetrics(self, difference):
+        """Add image QA metrics to the Task metadata.
+
+        Parameters
+        ----------
+        difference : `lsst.afw.image.Exposure`
+            The target image to calculate metrics for.
+        """
+        mask = difference.mask
+        goodPix = (mask.array & mask.getPlaneBitMask(self.config.detection.excludeMaskPlanes)) == 0
+        self.metadata.add("nGoodPixels", np.sum(goodPix))
+        self.metadata.add("nBadPixels", np.sum(~goodPix))
+        detPosPix = (mask.array & mask.getPlaneBitMask("DETECTED")) > 0
+        detNegPix = (mask.array & mask.getPlaneBitMask("DETECTED_NEGATIVE")) > 0
+        self.metadata.add("nPixelsDetectedPositive", np.sum(detPosPix))
+        self.metadata.add("nPixelsDetectedNegative", np.sum(detNegPix))
+        detPosPix *= ~goodPix
```

Comment:
Took me a sec to realize you meant `detPosPix &= ~goodPix`. 

What you have works, but was slower for this reader to read.